packages/react-query/src/types.ts

```diff
@@ -96,6 +116,11 @@ export type DefinedUseInfiniteQueryResult<
   TError = DefaultError,
 > = DefinedInfiniteQueryObserverResult<TData, TError>
 
+export type UseSuspenseInfiniteQueryResult<
+  TData = unknown,
+  TError = DefaultError,
+> = Omit<InfiniteQueryObserverSuccessResult<TData, TError>, 'isPlaceholderData'>
```

Comment:
I thought If we don't have `enabled` option and have default `suspense` option, we don't have to use `DefinedUseInfiniteQueryResult` containing unnecessary `InfiniteQueryObserverRefetchErrorResult`.

```ts
type DefinedInfiniteQueryObserverResult<TData = unknown, TError = DefaultError> = InfiniteQueryObserverRefetchErrorResult<TData, TError> | InfiniteQueryObserverSuccessResult<TData, TError>;
```

We just need only InfiniteQueryObserverSuccessResult for useSuspenseInfiniteQuery.
So we could remove `isSuccess: false`, `error`, `isError: true` etc. to do  type-narrow

```ts
interface InfiniteQueryObserverRefetchErrorResult<TData = unknown, TError = DefaultError> extends InfiniteQueryObserverBaseResult<TData, TError> {
    data: TData;
    error: TError;
    isError: true;
    isPending: false;
    isLoadingError: false;
    isRefetchError: true;
    isSuccess: false;
    status: 'error';
}
interface InfiniteQueryObserverSuccessResult<TData = unknown, TError = DefaultError> extends InfiniteQueryObserverBaseResult<TData, TError> {
    data: TData;
    error: null;
    isError: false;
    isPending: false;
    isLoadingError: false;
    isRefetchError: false;
    isSuccess: true;
    status: 'success';
}
```

This change is needed by same reason with https://github.com/TanStack/query/pull/5755#discussion_r1269836999